cmd/bosun/expr/elastic.go

```diff
@@ -238,17 +240,42 @@ func ESLTE(e *State, T miniprofiler.Timer, key string, lte float64) (*Results, e
 // ElasticHosts is an array of Logstash hosts and exists as a type for something to attach
 // methods to.  The elasticsearch library will use the listed to hosts to discover all
 // of the hosts in the config
-type ElasticHosts []string
+// type ElasticHosts []string
+type ElasticHosts struct {
+	Hosts     map[string]ElasticConfig
+	PrefixKey string
+	Keys      []string
+}
+
+type ElasticConfig struct {
+	Hosts             []string
+	SimpleClient      bool
+	ClientOptionFuncs []elastic.ClientOptionFunc
+}
 
 // InitClient sets up the elastic client. If the client has already been
 // initalized it is a noop
 func (e ElasticHosts) InitClient() error {
-	if esClient == nil {
-		var err error
-		esClient, err = elastic.NewClient(elastic.SetURL(e...), elastic.SetMaxRetries(10))
-		if err != nil {
-			return err
-		}
+	var err error
+	if e.PrefixKey != "" {
+		slog.Infof("es prefix key found connecting to host: %s", e.Hosts[e.PrefixKey])
```

Comment:
this should be removed since it logs whenever the expressions are executed so it will be spammy.